### PR TITLE
Fix #5184, ensure Typescript developer experience doesn't influence user experience

### LINKED ISSUE #5184: master branch build broken: 'npm i knex/knex@master' does not work
# Environment

Knex version: current master branch
OS: Windows

# Bug

1. Cant install from newer commits after Tag 2.0.0. Seems to be broken after dependerBot.

2. Error message

`npm i knex/knex#2.0.0`
image
this works.

`npm i knex/knex#master`
image
this does not work anymore.

I will try to update you, which commit is causing this issue. My first assumption is the dependerBot. Thanks in advance 👍

### PR DESCRIPTION
**Pull request by code-ape**
## Related PRs and Issues

1. #5184
2. #5151

## Reason

#5151 caused all end user installs of Knex to fail due to single line in `package.json`. Intention was that when developers ran `npm install` while working on Knex it would automatically build the Typescript files. This was a mistake and it should be a Husky script for the git hook `post-checkout`.

## Changes

1. Converted automation to build Typescript files from npm `postInstall` and `prepublishOnly` hooks to `prepare` hook in `package.json`.
2. Added Husky git hook to automatically build Typescript files on branch changes in `package.json`.
3. Fixed `files` section to properly exclude non-publishable `lib/` files in `package.json`.
4. Fixed npm command `build:gitignore` to work on Windows in `package.json`.
5. Updated contributing guide to mirror above changes in `CONTRIBUTING.md`.
6. Added Github Actions CI integration test for installing the package on all combinations of OS and Node version in `.github/workflows/integration-tests.yml`.
7. Updated Husky dependency from 7.x to 8.x in `package.json`.
8. Updated Husky pre-commit hook with newer shell template in `.husky/pre-commit`.
9. Added entry to ignore packed Knex package tar gzip files in `.gitignore`.

### CODE REVIEW

**Comment by code-ape** [comment]
@kibertoad please have all tests run and if they pass I'll update description and convert from draft for review!

**Comment by kibertoad** [comment]
@code-ape integration tests got nuked somehow

**Comment by code-ape** [comment]
@kibertoad it's because I'm an outside contributor and changed that CI file. I think there should be some way for you to "approve and run it"?

**Comment by kibertoad** [comment]
@code-ape I think restriction only applies to first-time committers, CI should kick-in automatically for you.

File: .github/workflows/integration-tests.yml [hunk 1]
```diff
@@ -91,3 +91,25 @@ jobs:
             --file "scripts/docker-compose.yml" \
             down
         if: matrix.database-type != 'sqlite3' && matrix.database-type != 'better-sqlite3'
+
+  build:
```
Comment (kibertoad) [inline .github/workflows/integration-tests.yml:95]: is this name not a duplicate with existing one?

Comment (code-ape) [reply .github/workflows/integration-tests.yml]: Good catch! Fixed and pushed.

**Comment by code-ape** [comment]
@kibertoad and @OlivierCavadenti, draft status removed, description updated to describe all changes, and all CI tests passing. Handing it off to you to do as you will 😄

**Review by OlivierCavadenti — APPROVED** [review]

**Review by kibertoad — APPROVED** [review]

**Pull request merged**